core/isolate.rs

```diff
@@ -174,6 +210,27 @@ impl Isolate {
     }
   }
 
+  extern "C" fn dyn_import(
+    user_data: *mut c_void,
+    specifier: *const c_char,
+    referrer: *const c_char,
+    id: deno_dyn_import_id,
+  ) {
+    assert_ne!(user_data, std::ptr::null_mut());
+    let isolate = unsafe { Isolate::from_raw_ptr(user_data) };
+    let specifier = unsafe { CStr::from_ptr(specifier).to_str().unwrap() };
+    let referrer = unsafe { CStr::from_ptr(referrer).to_str().unwrap() };
+    debug!("dyn_import specifier {} referrer {} ", specifier, referrer);
+
+    if let Some(ref f) = isolate.config.dyn_import {
+      let inner = f(specifier, referrer);
+      let fut = DynImport { inner, id };
+      isolate.pending_dyn_imports.push(fut);
```

Comment:
You need to poll the freshly created future, otherwise the task won't be woken up and you rely on the event look ticking coincidentally because of other ops.
See the logic regarding `have_unpolled_ops`.

Reply:
I added `task::current().notify()` here - does that work?

Reply:
That works.